Sources/hyper-focus/action_handler.swift

```diff
@@ -20,10 +20,16 @@ enum ActionHandler {
     }
 
     static func appAction(_ data: SwitchingActivity) -> Bool {
-        if data.configuration.block_apps.contains(data.app) {
+        if match(data.app, data.configuration.block_apps) {
+            if match(data.app, data.configuration.allow_apps) {
+                log("app is in allow_apps, releasing")
+                return false
+            }
             log("app is in block_apps, hiding application to prevent usage")
             // TODO: sometimes this hide method does not work
-            NSWorkspace.shared.frontmostApplication!.hide()
+            //NSWorkspace.shared.frontmostApplication!.hide()
+            NSWorkspace.shared.frontmostApplication!.terminate()
```

Comment:
Note that I did this because hiding wasn't working for some classes of apps. I'm not familiar with the terminology, but it seems like it might be apps like raycast or Alfred that open a panel without having a dock icon.

Terminating them does work, which is why I switched to it in my local repo, but I'm guessing you might not want this. A setting might be better.

Reply:
Could you explain your use case here?

Why would you want raycast/alfred in a blocklist?

The assumption I've made is applications which you can't hide, probably aren't the type of applications that are distracting.

Which applications are distracting that you can't hide?

Reply:
It might be a good assumption indeed. In my specific case, it was mostly because during testing I found that the [Shortery app](https://apps.apple.com/it/app/shortery/id1594183810?mt=12) wasn't being hidden.
I have a habit of messing with automation stuff more than I should, so I decided that I may have wanted to block it anyway, and thus made that change.

In any case, I have now reverted it even locally because, as you probably already realized, quitting apps leads to possible data loss, and that outweighs the minor benefits. I'll revert it here too.